storage/src/store/mod.rs

```diff
@@ -0,0 +1,919 @@
+//!         locations_journal_partition: "locations_partition".to_string(),
+//!         locations_items_per_blob: 4,
+//!         translator: TwoCap,
+//!         buffer_pool: PoolRef::new(PAGE_SIZE, PAGE_CACHE_SIZE),
+//!     };
+//!     let mut store =
+//!         Store::<_, Digest, Digest, TwoCap>::init(ctx.with_label("store"), config)
+//!             .await
+//!             .unwrap();
+//!
+//!     // Insert a key-value pair
+//!     let k = Digest::random(&mut ctx);
+//!     let v = Digest::random(&mut ctx);
+//!     store.update(k, v).await.unwrap();
+//!
+//!     // Fetch the value
+//!     let fetched_value = store.get(&k).await.unwrap();
+//!     assert_eq!(fetched_value.unwrap(), v);
+//!
+//!     // Commit the operation to make it persistent
+//!     store.commit().await.unwrap();
+//!
+//!     // Delete the key's value
+//!     store.delete(k).await.unwrap();
+//!
+//!     // Fetch the value
+//!     let fetched_value = store.get(&k).await.unwrap();
+//!     assert!(fetched_value.is_none());
+//!
+//!     // Commit the operation to make it persistent
+//!     store.commit().await.unwrap();
+//!
+//!     // Destroy the store
+//!     store.destroy().await.unwrap();
+//! });
+//! ```
+
+use crate::{
+    index::Index,
+    journal::{
+        fixed::{Config as FConfig, Journal as FJournal},
+        variable::{Config as VConfig, Journal as VJournal},
+        Error as JError,
+    },
+    translator::Translator,
+};
+use commonware_codec::Read;
+use commonware_runtime::{buffer::PoolRef, Clock, Metrics, Storage as RStorage};
+use commonware_utils::{sequence::U32, Array, Span};
+use futures::{pin_mut, try_join, StreamExt};
+use tracing::{debug, warn};
+
+pub mod operation;
+use operation::Operation;
+
+/// The size of the read buffer to use for replaying the operations log when rebuilding the
+/// snapshot.
+const SNAPSHOT_READ_BUFFER_SIZE: usize = 1 << 16;
+
+/// Configuration for initializing a [`Store`] database.
+#[derive(Clone)]
+pub struct Config<T: Translator, C> {
+    /// The name of the [`RStorage`] partition used to persist the log of operations.
+    pub log_journal_partition: String,
+
+    /// The size of the write buffer to use for each blob in the [`VJournal`].
+    pub log_write_buffer: usize,
+
+    /// Optional compression level (using `zstd`) to apply to log data before storing.
+    pub log_compression: Option<u8>,
+
+    /// The codec configuration to use for encoding and decoding log items.
+    pub log_codec_config: C,
+
+    /// The number of operations to store in each section of the [`VJournal`].
+    pub log_items_per_section: u64,
+
+    /// The name of the [`RStorage`] partition used for the location map.
+    pub locations_journal_partition: String,
+
+    /// The number of items to put in each blob in the location map.
+    pub locations_items_per_blob: u64,
+
+    /// The [`Translator`] used by the compressed index.
+    pub translator: T,
+
+    /// The buffer pool to use for caching data.
```

Comment:
I wonder if it worth added a TODO in `//` below this comment (or in the init function) to use this for the variable journal too?

Hopefully, we can get this in soon: https://github.com/commonwarexyz/monorepo/issues/1223